modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpPipeliningHandler.java

```diff
@@ -198,6 +200,8 @@ private void doWrite(ChannelHandlerContext ctx, Netty4HttpResponse readyResponse
             doWriteFullResponse(ctx, fullResponse, promise);
         } else if (readyResponse instanceof Netty4ChunkedHttpResponse chunkedResponse) {
             doWriteChunkedResponse(ctx, chunkedResponse, promise);
+        } else if (readyResponse instanceof Netty4ChunkedHttpContinuation chunkedContinuation) {
+            doWriteChunkedContinuation(ctx, chunkedContinuation, promise);
```

Comment:
I wonder whether we need to assert that there was a `Netty4ChunkedHttpResponse` preceeds the continuation. for with the same sequence to ensure we won't get the continuation as the first part of a response?

Reply:
The code structure prevents this I think (i.e. the continuation listener for a particular `writeSequence` is not even created until sending the initial `Netty4ChunkedHttpResponse`, and is wrapped in an `assertOnce`. I added one more assertion in 4954c8e2fdd to help document how this works but I don't see an easy way to do exactly what you suggest.

Reply:
It's likely me overthinking about this. I agree the code itself here is fine. I was thinking that if an API responds with a Continuation first, it could be helpful to catch the issue (of misuse). This comment is related to the other one where I am suggesting reducing the visibilty, also to prevent misuse.

Reply:
This is all hidden in the `transport-netty4` module which means it's already pretty much inaccessible (indeed invisible) to everyone else. I don't think we need more protection than that really.